MapboxCoreNavigation/RouteController.swift

```diff
@@ -236,20 +236,30 @@ extension RouteController: CLLocationManagerDelegate {
             if !isCloseToCurrentStep && isCloseToUpComingStep {
                 
                 // Increment the step
-                routeProgress.currentLegProgress.stepIndex += 1
```

Comment:
@1ec5 i wonder if this is the root cause of https://github.com/mapbox/mapbox-navigation-ios/pull/255